js/app/courseHistoryStore.js

```diff
@@ -82,17 +82,12 @@ const CourseHistoryStore = new GObject.Class({
             tags_match_all: ['EknSetObject'],
             sort: Eknc.QueryObjectSort.SEQUENCE_NUMBER,
         });
-        Eknc.Engine.get_default().query(query, null, (engine, task) => {
-            let results;
-            try {
-                results = engine.query_finish(task);
-            } catch (error) {
-                logError(error);
-                return;
-            }
+        Eknc.Engine.get_default().query_promise(query).then((results) => {
```

Comment:
Ah misread, are you sure we decided on straight ruler and not
```
Eknc.Engine.get_default().query_promise(query)
    .then(results => {
        blah blah
    })
    .catch(error => {
        logError(error)
    });
```
Fine with whatever, but don't want to rearrange the style here more than once so I'll wait till consensus to touch this.